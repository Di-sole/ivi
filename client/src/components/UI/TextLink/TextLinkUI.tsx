import Link from 'next/link';
import styles from './TextLinkUI.module.scss';

interface TextLinkUIProps {
  href: string;
  children: React.ReactNode;
  option: 'bright' | 'dim' | 'gradient';
}

const TextLinkUI = ({ href, children, option }: TextLinkUIProps) => {
  return (
    <Link href={href} className={[styles.link, styles[`link_${option}`]].join(' ')}>
      {children}
    </Link>
  );
};

export default TextLinkUI;
