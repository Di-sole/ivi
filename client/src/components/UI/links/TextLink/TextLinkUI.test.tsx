import { render, screen } from '@testing-library/react';
import TextLinkUI from './TextLinkUI';

describe('TEXTLINKUI SNAPSHOTS TESTS', () => {
    test('bright TEXTLINKUI should not be changed', () => {
        render(
            <TextLinkUI href="/" option="bright">
                Ссылка
            </TextLinkUI>,
        );
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });

    test('dim TEXTLINKUI should not be changed', () => {
        render(
            <TextLinkUI href="/" option="dim">
                Ссылка
            </TextLinkUI>,
        );
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });

    test('gradient TEXTLINKUI should not be changed', () => {
        render(
            <TextLinkUI href="/" option="gradient">
                Ссылка
            </TextLinkUI>,
        );
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });
});
