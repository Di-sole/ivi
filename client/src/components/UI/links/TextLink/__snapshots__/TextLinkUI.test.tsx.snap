// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEXTLINKUI SNAPSHOTS TESTS bright TEXTLINKUI should not be changed 1`] = `
<a
  class="container container_bright"
  href="/"
>
  Ссылка
</a>
`;

exports[`TEXTLINKUI SNAPSHOTS TESTS dim TEXTLINKUI should not be changed 1`] = `
<a
  class="container container_dim"
  href="/"
>
  Ссылка
</a>
`;

exports[`TEXTLINKUI SNAPSHOTS TESTS gradient TEXTLINKUI should not be changed 1`] = `
<a
  class="container container_gradient"
  href="/"
>
  Ссылка
</a>
`;
